mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -2139,34 +2139,34 @@ def NVVM_WgmmaFenceAlignedOp : NVVM_PTXBuilder_Op<"wgmma.fence.aligned"> {
     [For more information, see PTX ISA](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#asynchronous-warpgroup-level-matrix-instructions-wgmma-fence)
   }];
   let assemblyFormat = "attr-dict";
-  let extraClassDefinition = [{
-    std::string $cppClass::getPtx() { return std::string("wgmma.fence.sync.aligned;"); }
+  string llvmBuilder = [{
+    createIntrinsicCall(builder, llvm::Intrinsic::nvvm_wgmma_fence_sync_aligned);
   }];
 }
 
-def NVVM_WgmmaGroupSyncAlignedOp : NVVM_PTXBuilder_Op<"wgmma.commit.group.sync.aligned">,
+def NVVM_WgmmaGroupSyncAlignedOp : NVVM_Op<"wgmma.commit.group.sync.aligned">,
   Arguments<(ins )> {
   let assemblyFormat = "attr-dict";
   let description = [{
     Commits all prior uncommitted warpgroup level matrix multiplication operations.
     
     [For more information, see PTX ISA](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#asynchronous-warpgroup-level-matrix-instructions-wgmma-commit-group)
   }];
-  let extraClassDefinition = [{
-    std::string $cppClass::getPtx() { return std::string("wgmma.commit_group.sync.aligned;"); }
+  string llvmBuilder = [{
+    createIntrinsicCall(builder, llvm::Intrinsic::nvvm_wgmma_commit_group_sync_aligned);
   }];
 }
 
-def NVVM_WgmmaWaitGroupSyncOp : NVVM_PTXBuilder_Op<"wgmma.wait.group.sync.aligned">{
-  let arguments = (ins I32Attr:$group);
+def NVVM_WgmmaWaitGroupSyncOp : NVVM_Op<"wgmma.wait.group.sync.aligned">{
+  let arguments = (ins I64Attr:$group);
```

Comment:
Does this data-type change (from I32 to I64) not need a corresponding change in the NVGPU->NVVM lowering?
(for NVGPU_DeviceAsyncWaitOp).

Reply:
I think `NVGPU_DeviceAsyncWaitOp` lowers to `NVVM_CpAsyncWaitGroupOp` so we may not need it there. But I see that the `NVGPU_WarpgroupMmaOp` which emits an `NVVM_WgmmaWaitGroupSyncOp` while lowering has the argument `waitGroup` which is I32 currently, so I have updated it to I64 now. Thanks!

Reply:
Yes, I meant the wgmma.waitgroup attr for wgmma Ops.

The tests (for wgmma using the waitgroup attr) don't need an update since they were using a direct constant, I believe?

Reply:
out of curiosity - why do you need to make this type i64? It cannot be that large.

Reply:
Oh, that's because the argument for [wgmma.wait_group](https://docs.nvidia.com/cuda/parallel-thread-execution/#asynchronous-warpgroup-level-matrix-instructions-wgmma-wait-group) defines the operand `N` to be an integer constant and [here](https://docs.nvidia.com/cuda/parallel-thread-execution/#integer-constants), they are defined to be always be 64-bits. So, the corresponding intrinsic also takes an `i64` so it needed to be updated here, although in this case I also think realistically it wouldn't be that large.

Reply:
My recollection was that `N` could be a maximum of 16. It looks like that’s not the case.